### PR TITLE
Added MediaInput

### PR DESCRIPTION
**Pull request by apilguk**

### CODE REVIEW

File: lib/program/session.js [hunk 1]
```diff
@@ -142,7 +142,7 @@ export default class Session {
 
   /**
    * @description Destroy all initialized operations,
-   * texture and outher data connected this session.
+   * texture and outer data connected this session.
```
Comment (WorldThirteen) [inline lib/program/session.js:145]:
```suggestion
   * textures and other data connected to this session.
```

File: lib/index.d.ts [hunk 2]
```diff
@@ -110,6 +110,7 @@ export function calcHAARFeature(input: Tensor, feature: number[], size: number,
 
 /* ops */
 
+export function mediaInput(input: MediaInputType): Tensor
```
Comment (WorldThirteen) [inline lib/index.d.ts:113]: There is no mediaInput op added.

File: lib/utils.js [hunk 3]
```diff
@@ -34,7 +34,10 @@ export const isValidShape = shape => Array.isArray(shape)
   && shape.length > 0
   && !shape.some(n => n % 1 !== 0);
 export const isOperation = op => op instanceof gm.Operation;
+export const isMediaInput = op => op instanceof gm.MediaInput;
 export const isTensor = tensor => tensor instanceof gm.Tensor;
+export const isVideoElement = input => input instanceof HTMLVideoElement;
```
Comment (WorldThirteen) [inline lib/utils.js:39]: Added functions don't have corresponded types in index.d.ts.

**Review by WorldThirteen — COMMENTED** [review]
1. `tensorFrom` util have an InputType as the first argument but its code doesn't support MediaInput as an argument, while `InputType` now contains MediaInput.
2. `generateTransformMatrix` have a similar story, but it should have a Tensor input toe for tMAtrix and not InputType. I mentioning this in this PR since InputType ts type changed here.

**Comment by apilguk** [comment]
tnx for the feedback

**Review by WorldThirteen — APPROVED** [review]
LGTM.

**Pull request merged**